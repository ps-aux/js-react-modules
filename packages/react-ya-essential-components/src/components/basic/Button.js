import React from 'react'

const Button = ({label, icon, ...props}) =>
    <button className="button-control" {...props}>
        {label}
    </button>

export default Button
