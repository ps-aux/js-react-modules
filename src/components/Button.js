import React from 'react'
import Text from './Text'
import { addClassName } from '../utils'

const cn = addClassName('button')

const Button = ({children, label, className, ...rest}) =>
    <button className={cn(className)} {...rest}>
        {children || <Text>{label}</Text>}
    </button>

export default Button
