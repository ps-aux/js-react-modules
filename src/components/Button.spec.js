import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

describe('Button renders', () => {
    it('children content properly', () => {
        const comp = renderer.create(
            <Button>Text</Button>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('label text', () => {
        const comp = renderer.create(
            <Button label="Text"/>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('additional className', () => {
        const comp = renderer.create(
            <Button label="any" className="additional"/>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
