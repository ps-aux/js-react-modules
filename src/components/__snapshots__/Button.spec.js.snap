// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders additional className 1`] = `
<button
  className="button additional"
>
  <span
    cn="text"
  >
    any
  </span>
</button>
`;

exports[`Button renders children content properly 1`] = `
<button
  className="button"
>
  Text
</button>
`;

exports[`Button renders label text 1`] = `
<button
  className="button"
>
  <span
    cn="text"
  >
    Text
  </span>
</button>
`;
